scripts/globals/dark_ixion.lua

```diff
@@ -462,3 +577,301 @@ xi.darkixion.onMobDisengage = function(mob)
     mob:setAggressive(false)
     mob:setLocalVar('StygianLanded', 0)
 end
+
+-- Disable cyclomatic complexity check for this function:
+-- luacheck: ignore 561
+xi.darkixion.onMobFight = function(mob, target)
+    -- Since its autos are technically TP moves, lets emulate WS selection here
+    if
+        ((mob:getTP() >= 2900 and mob:getHPP() > 66) or
+        (mob:getTP() >= 1900 and mob:getHPP() > 33) or
+        (mob:getTP() >= 900 and mob:getHPP() > 0)) and
+        mob:getLocalVar('timeSinceWS') < os.time() - 3 and
+        mob:checkDistance(target) < 15 and
+        mob:getLocalVar('charging') == 0 and
+        mob:actionQueueEmpty()
+    then
+        mob:setLocalVar('timeToWS', 1)
+    end
```

Comment:
At a glance, this condition is awful, can you find a way to break it up?

Reply:
I'm essentially reinventing the core logic that decides if a mob should use a mobskill. I could definitely collapse the top three into some continuous function, but imo the rest are needed.